src/include/distributed/metadata_sync.h

```diff
@@ -0,0 +1,41 @@
+/*-------------------------------------------------------------------------
+ *
+ * metadata_sync.h
+ *	  Type and function declarations used to sync metadata across all
+ *	  workers.
+ *
+ * Copyright (c) 2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef METADATA_SYNC_H
+#define METADATA_SYNC_H
+
+
+#include "nodes/pg_list.h"
+
+
+/* Functions declarations for metadata syncing */
+extern bool ShouldSyncTableMetadata(Oid relationId);
+extern List * MetadataCreateCommands(void);
+extern List * ClusterMetadataDropCommands(void);
+extern char * DistributionCreateCommand(Oid distributedRelationId,
+										char distributionMethod,
+										char *distributionColumnName);
+extern char * DistributionDeleteCommand(char *schemaName,
+										char *tableName);
+extern char * TableOwnerResetCommand(Oid distributedRelationId);
+extern char * NodeListInsertCommand(List *workerNodeList);
+extern List * ShardListInsertCommand(List *shardIntervalList);
+extern char * NodeDeleteCommand(uint32 nodeId);
+extern List * GetDistributedTableDDLEvents(Oid relationId, char distributionMethod,
+										   char *distributionColumn);
+
+
+#define DELETE_ALL_NODES "TRUNCATE pg_dist_node"
+#define REMOVE_ALL_CLUSTERED_TABLES_COMMAND \
+	"SELECT worker_drop_distributed_table(logicalrelid) FROM pg_dist_partition WHERE iscluster"
```

Comment:
Hmz, iscluster no longer exists. I suppose we could just drop all distributed tables on the worker, since the user cannot create any additional ones.